main.py

```diff
@@ -602,6 +695,8 @@ def _install_flows(
 
                 try:
                     self._send_flow_mod(flow.switch, flow_mod)
+                    if send_barrier:
+                        self._send_barrier_request(flow.switch, flow_mod)
```

Comment:
@viniarck my previous comment is based on this code. From what I understood, you will send a barrier request here and return to the user with success, correct? Later on, the barrier reply will be received and will persist the flow. The question is: what if something happens to Kytos in the mean time? For instance, if the Kytos server gets restarted right after receiving a successful return code from a flow mode rest request (not necessarily meaning the barrier reply was received and, consequently, the flow was not persisted), the user request won't be recorded. What do you think?

Reply:
That's correct @italovalcy. It's being sent the same way we send the flow mods and catching the connection error exception, if any. For this case, like we have for flow mods, it could happen as we are writing in the DB storage via async threaded event (although kytos tries to shutdown gracefully, so, sometimes that might also complete depending on the signal or how it was terminated), so depending on at which instant it restart it could lead to this case you're describing.

Reply:
All in all, I think we'd be able to solve this canonically once we prioritize an integration with another storage backed that supports transactions, writing synchronously during the request (or event) cycle since a restart could happen at any time, and then only resulting in a success if the transaction succeeded. Let me know if you see any other solution in the meantime, I suppose we should map this requirement as an issue, wdyt?

Reply:
Great! Understood and agreed :)